packages/cli/docs-resolver/src/visitNavigationAst.ts

```diff
@@ -48,6 +48,19 @@ export async function visitNavigationAst({
                             });
                         })
                     );
+                } else if ("variants" in tab && tab.variants != null) {
+                    await Promise.all(
+                        tab.variants.flatMap((variant) =>
+                            variant.layout.map(async (item: docsYml.RawSchemas.NavigationItem) => {
+                                await visitNavigationItem({
+                                    navigationItem: item,
+                                    visitor,
+                                    apiWorkspaces,
+                                    context
+                                });
+                            })
+                        )
+                    );
```

Comment:
can you add a helper function that is a type guard instead of using the `in` operator?